cpp/lib/ClientConfigParser.cpp

```diff
@@ -0,0 +1,281 @@
+  if (homeDir && strlen(homeDir) != 0)
+  {
+    homeDirFilePath = std::string(homeDir) + PATH_SEP + SF_CLIENT_CONFIG_FILE_NAME;
+  } else {
+    // USERPROFILE is empty, try HOMEDRIVE and HOMEPATH
+    char envbuf2[MAX_PATH + 1];
+    char* homeDriveEnv = sf_getenv_s("HOMEDRIVE", envbuf, sizeof(envbuf));
+    char* homePathEnv = sf_getenv_s("HOMEPATH", envbuf2, sizeof(envbuf2));
+    if (homeDriveEnv && strlen(homeDriveEnv) != 0 && homePathEnv && strlen(homePathEnv) != 0)
+    {
+      homeDirFilePath = std::string(homeDriveEnv) + homePathEnv + PATH_SEP + SF_CLIENT_CONFIG_FILE_NAME;
+    }
+  }
+  if (boost::filesystem::is_regular_file(homeDirFilePath))
+  {
+    CXX_LOG_INFO("Using client configuration path from home directory: %s", homeDirFilePath.c_str());
+    return homeDirFilePath;
+  }
+#else
+  char* homeDir;
+  if ((homeDir = sf_getenv_s("HOME", envbuf, sizeof(envbuf))) && (strlen(homeDir) != 0))
+  {
+    std::string homeDirFilePath = std::string(homeDir) + PATH_SEP + SF_CLIENT_CONFIG_FILE_NAME;
+    if (boost::filesystem::is_regular_file(homeDirFilePath))
+    {
+      CXX_LOG_INFO("Using client configuration path from home directory: %s", homeDirFilePath.c_str());
+      return homeDirFilePath;
+    }
+  }
+#endif
+  return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+sf_bool EasyLoggingConfigParser::parseConfigFile(
+  const std::string& in_filePath,
+  client_config& out_clientConfig)
+{
+  value jsonConfig;
+  std::string err;
+  std::ifstream configFile;
+  configFile.open(in_filePath, std::fstream::in | std::ios::binary);
+  if (!configFile)
+  {
+    CXX_LOG_INFO("Could not open a file. The file may not exist: %s",
+      in_filePath.c_str());
+    return false;
+  }
+#if !defined(WIN32) && !defined(_WIN64)
+  if (!checkIfValidPermissions(in_filePath))
+  {
+    return false;
+  }
+#endif
+  err = parse(jsonConfig, configFile);
+
+  if (!err.empty())
+  {
+    CXX_LOG_ERROR("Error in parsing JSON: %s, err: %s", in_filePath.c_str(), err.c_str());
+    return false;
+  }
+
+  value commonProps = jsonConfig.get("common");
+  checkUnknownEntries(commonProps);
+  if (commonProps.is<object>())
+  {
+    if (commonProps.contains("log_level") && commonProps.get("log_level").is<std::string>())
+    {
+      const char* logLevel = commonProps.get("log_level").get<std::string>().c_str();
+      size_t logLevelSize = strlen(logLevel) + 1;
+      out_clientConfig.logLevel = (char*)SF_CALLOC(1, logLevelSize);
+      sf_strcpy(out_clientConfig.logLevel, logLevelSize, logLevel);
+    }
+    if (commonProps.contains("log_path") && commonProps.get("log_path").is<std::string>())
+    {
+      const char* logPath = commonProps.get("log_path").get<std::string>().c_str();
+      size_t logPathSize = strlen(logPath) + 1;
+      out_clientConfig.logPath = (char*)SF_CALLOC(1, logPathSize);
+      sf_strcpy(out_clientConfig.logPath, logPathSize, logPath);
+    }
+  }
+  return true;
```

Comment:
The reason it still returns true is because if the user passes the log level but not log path to log_init() and the config file only has log path, we would still like the logger to log to the log path set in the config (and vice versa).